sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -319,23 +321,14 @@ def database_identifier(self):
         # type: () -> str
         if self._database_identifier is None:
             template = Template(self._config.database_identifier.get('template') or '$resolved_hostname')
-            tag_dict = {}
-            tags = self.tags.copy()
-            # sort tags to ensure consistent ordering
-            tags.sort()
-            for t in tags:
-                if ':' in t:
-                    key, value = t.split(':', 1)
-                    if key in tag_dict:
-                        tag_dict[key] += f",{value}"
-                    else:
-                        tag_dict[key] = value
+            # Copy self.tag_manager._tags and map values to single values instead of lists
+            tag_dict = {k: v[0] if isinstance(v, list) and v else v for k, v in self.tag_manager._tags.items()}
```

Comment:
Hm in mysql i just ended up leaving the existing logic here and using `tags = self.tag_manager.get_tags()`. I don't love accessing the internal tags object directly already. If we want to leave this as is I'm fine for now to avoid package bump toil, but we should probably update TagManager to expose this nicely.

Reply:
Yeah, agreed we should just expose it cleanly. The overheads not too bad so I'll revert to keep it consistent